sdk/storage/azure-storage-blobs/CMakeLists.txt

```diff
@@ -41,6 +41,7 @@ endif()
 
 set(
   AZURE_STORAGE_BLOBS_HEADER
```

Comment:
I was under the impression that `XXX_HEADER` should include all `.hpp` and `XXX_SOURCE` should include `.cpp`.

what is the standard when we decide a file should be categorized as `HEADER` or `SOURCE`?

Reply:
I've seen it being done both ways, even inside storage.
By-directory is more popular than by-extension.
I think ultimately, it does not have any difference.
If you have any preference, or want me to leave them in Storage just as they were (although it wasn't consistent even there), let me know.
Technically, by-directory is easier to automate(`ls dir -Recurse -Name -File | % {$_.replace("\","/")}`), however doing it by-extension shouldn't be too hard to implement either. Not that I am planning to integrate this automation - I just did it semi-manually and I hope it will be good for another year or so.

Reply:
> I've seen it being done both ways, even inside storage.

Let's work to make this consistent and decide the pattern we want to follow here.

Reply:
I don't completely understand, we are picking one approach here, which is by-directory.
Codegen does not produce cmakelists.txt files.
If anything, I would rather drop the separation at all, and just use single list - it does not we use that distinction in any meaningful way anywhere, we just combine them at the earliest convenience.
I would not even declare a variable like we do today - just feed all the files to CMake `add_library()` call - that variable with a list of files is not used anywhere else anyways.
We can briefly discuss this topic during our C++ sync meeting.

But I think we should just go with what `cmake-format` can offer us, plus look into what VisualStudio and VSCode support by default (they can add files once you add them inside an IDE, it is just currently we never accept what they generate because they generate paths with quotes and wrong indentation; but all that may change once we integrate with cmake-format).

So for that reason, I think the issue tracking this that you're looking into is the #5066. Whichever the auto-format (with avaliable config tweaks) can produce for us - that's the format that we should use. The rest does not matter that much.